tests/scenario/test_charm.py

```diff
@@ -33,18 +34,42 @@ def test_start(context, base_state):
 def test_tempo_restart_on_ingress_v2_changed(context, tmp_path, requested_protocol):
     # GIVEN
     # an initial configuration with an otlp_http receiver
-    tempo_config = tmp_path / "tempo.yaml"
+    container, tempo = _tempo_mock_with_initial_config(tmp_path)
+
+    # the remote end requests an otlp_grpc endpoint
+    ingress = Relation(
+        "tracing", remote_app_data=TracingRequirerAppData(receivers=[requested_protocol]).dump()
+    )
+
+    # WHEN
+    # the charm receives an ingress(v2) relation-changed requesting an otlp_grpc receiver
+    state = State(leader=True, containers=[tempo], relations=[ingress])
+    context.run(ingress.changed_event, state)
+
+    # THEN
+    # Tempo pushes a new config to the container filesystem
+    fs = tempo.get_filesystem(context)
+    cfg_path = Path(str(fs) + Tempo.config_path)
+    new_config = yaml.safe_load(cfg_path.read_text())
```

Comment:
I imagine that this sequence will be repeated, in which case, would you consider:
- either moving it to a fixture
- or providing a convenience method or function?

Reply:
good point, done